app/components/image-viewer.ts

```diff
@@ -68,4 +88,65 @@ export default class ImageViewerComponent extends Component<ImageViewerArgs> {
           layers: [image],
         });
     });
+
+    exportTask = task(
+      async (): Promise<void> => {
+        if (!this.map) throw new Error('Map was not loaded yet.');
+        const mapCanvas = document.createElement('canvas');
+        if (this.width) mapCanvas.width = (this.width) as number;
+        if (this.height) mapCanvas.height = (this.height) as number;
+        const mapContext = mapCanvas.getContext('2d');
+        const canvas = this.mapCanvas;
+
+        if (!(canvas instanceof HTMLCanvasElement)) {
+          throw new Error('Canvas not loaded yet.');
+        }
+
+        let matrix: null | [number, number, number, number, number, number] =
+          null;
+        const transform = canvas.style.transform;
+
+        if (canvas.width > 0) {
+          matrix = transform
+            // eslint-disable-next-line no-useless-escape
+            .match(/^matrix\(([^\(]*)\)$/)?.[1]
+            ?.split(',')
+            .map((x) => Number(x)) as [
+            number,
+            number,
+            number,
+            number,
+            number,
+            number
+          ];
+        } else {
+          matrix = [
+            parseFloat(canvas.style.width) / canvas.width,
+            0,
+            0,
+            parseFloat(canvas.style.height) / canvas.height,
+            0,
+            0,
+          ];
+        }
+
+        if (!matrix) throw new Error('Could not export (matrix)');
+        const transformMatrix = new DOMMatrix(matrix.map((i) => i));
+        mapContext?.setTransform(transformMatrix);
+        mapContext?.drawImage(canvas, 0, 0, 500, 500);
```

Comment:
```suggestion
        mapContext?.drawImage(canvas, 0, 0, 500, 500);
```
have to be dynamic else it breaks in mobile !!!!